File: pset6/vigenere.py
import cs50
import sys

def main():
    if len(sys.argv) != 2:
        print("Usage: python ./vigenere.py k")
        exit(1)

    # check input and build the key array at the same time
    asciikey = []
    for c in sys.argv[1]:
        if c.isalpha() == False:
            print("Usage: python ./vigenere.py k")
            exit(1)
        else:
            asciikey.append(ord(c.lower()) - ord("a"))
    #print("asciikey: {}".format(asciikey))

    # get user input
    print("plaintext: ", end="")
    pt = cs50.get_string()

    #variable to keep track of the alpha character
    alphaplace = 0
    print("ciphertext: ", end="")
    for p in pt:
        if p.isalpha() == True:
            # if alpha character, encrypt

            if ord(p) >= ord("a") and ord(p) <= ord("z"):
                # if lowercase
                if ord(p) + asciikey[alphaplace % len(asciikey)] > ord("z"):
                    # wrap around if needd
                    print("{}".format(chr(ord("a") + 1 + ord(p) + asciikey[alphaplace % len(asciikey)] - ord("z"))), end = "")
                else:
                    print("{}".format(chr(ord(p) + asciikey[alphaplace % len(asciikey)])), end = "")
            elif ord(p) >= ord("A") and ord(p) <= ord("Z"):
                # if uppercase
                if ord(p) + asciikey[alphaplace % len(asciikey)] > ord("Z"):
                    #wrap around if needed
                    print("{}".format(chr(ord("A") + 1 + ord(p) + asciikey[alphaplace % len(asciikey)] - ord("Z"))), end = "")
                else:
                    print("{}".format(chr(ord(p) + asciikey[alphaplace % len(asciikey)])), end = "")

            alphaplace += 1
        else:
            # not alpha, don't encrypt
            print("{}".format(p), end = "")

    print()

    exit(0)


if __name__ == "__main__":
    main()

File: pset2/vigenere/vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    // get key from command line
    if (argc != 2)
    {
        printf("Usage: ./vigenere k\n");
        return 1;
    }

    string key = argv[1];
    for (int i = 0, n = strlen(key); i < n; i++)
    {
        if (isalpha(key[i]) == 0 )
        {
            printf("Usage: ./vigenere k\n");
            return 1;
        }
    }

    int keylength = strlen(key);

    // create the array of the key's ascii vals
    char asciikey[keylength];
    for (int i = 0; i < keylength; i++)
    {
        asciikey[i] = (int)tolower(key[i]) - 'a';

        //printf("asciikey[%i]=%i\n", i, asciikey[i]);
    }

    // get plaintext
    printf("plaintext: ");
    string pt = get_string();

    // variable to keep track of the alpha caracter
    int alphaplace = -1;
    // encipher
    printf("ciphertext: ");
    for (int i = 0, n = strlen(pt); i < n; i++)
    {
        if (isalpha(pt[i]))
        {
            alphaplace += 1;
            //printf("pt[%i]=%c; (int)\'%c\'=%i, mod of %i=%i, key[mod of %i]=%c, %i ----- ",
            //i, pt[i], pt[i], (int)pt[i], alphaplace, alphaplace % keylength, alphaplace % keylength, key[alphaplace % keylength], (int)asciikey[alphaplace % keylength]);

            if (pt[i] >= 'a' && pt[i] <= 'z')
            {
                if ((int)pt[i] + asciikey[alphaplace % keylength] > (int)'z')
                {
                    printf("%c", ('a' - 1) + (int)pt[i] + asciikey[alphaplace % keylength] - 'z');
                }
                else
                {
                    printf("%c", (int)pt[i] + asciikey[alphaplace % keylength]);
                }
            }
            else if (pt[i] >= 'A' && pt[i] <= 'Z')
            {
                if ((int)pt[i] + asciikey[alphaplace % keylength] > (int)'Z')
                {
                    printf("%c", ('A' - 1) + (int)pt[i] + asciikey[alphaplace % keylength] - 'Z');
                }
                else
                {
                    printf("%c", (int)pt[i] + asciikey[alphaplace % keylength]);
                }
            }
        }
        else
        {
            printf("%c", pt[i]);
        }
    }

    printf("\n");

    // print ciphertext
}